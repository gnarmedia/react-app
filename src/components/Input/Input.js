/**
 * @file Input component
 */

import React from 'react'

/**
 * Renders Input component.
 *
 * @example
 * <Input name="email" type="email" required />
 * <Input name="password" type="password" required />
 *
 * @param {*} props - Props.
 *
 * @returns {Function} Input component.
 */
const Input = props => (
  <div className="field">
    <div className="control">
      <input className="input is-large" {...props} />
    </div>
  </div>
)

Input.defaultProps = {
  placeholder: null,
  type: 'text',
}

export default Input
