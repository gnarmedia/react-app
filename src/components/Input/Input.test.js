/**
 * @file Input component tests
 */

import React from 'react'
import { render } from 'enzyme'

import Input from './Input'

describe('index', () => {
  it('should render the Input component', () => {
    const wrapper = render(<Input />)

    expect(wrapper).toMatchSnapshot()
  })
})
