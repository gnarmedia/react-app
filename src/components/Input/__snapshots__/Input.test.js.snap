// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index should render the Input component 1`] = `
<div
  class="field"
>
  <div
    class="control"
  >
    <input
      class="input is-large"
      type="text"
    />
  </div>
</div>
`;
